### DISCUSSION TITLE
如何配置headerDepth使其生效？

### DISCUSSION THREAD

**Discussion by hellobambi** [vuepress-theme-hope/vuepress-theme-hope#2930] [Q&A] [answered]
I have read the docs in https://vuepress-theme-hope.github.io/v2/zh/guide/layout/sidebar.html#%E5%B5%8C%E5%A5%97%E7%9A%84%E6%A0%87%E9%A2%98%E9%93%BE%E6%8E%A5
and config in the md page: 
`---
headerDepth: 5
---
`
But h4 is still not displayed. I noticed that maybe because
![image](https://user-images.githubusercontent.com/52453727/226278538-295d2b94-8364-4324-a8cd-8cc9686d9ec2.png)

But I don't know how to specific configuration to change markdown.extractHeaders.level and then makes headerDepth: 5 effect

Could you provide the specific steps? thanks for your help and support!

**Comment by Mister-Hope** [comment, marked as answer]
https://github.com/vuepress-theme-hope/vuepress-theme-hope/commit/bf84df5464f8d938bef5dda45ffc27e9d2816c58

**Reply by hellobambi** [reply]
Thank you for your reply Mr. Hope, and I have successfully used it! By the way, I noticed that **the hyperlinks on your site don't actually go to the "markdown.headers" section of vuePress**. This is because, your link is:
"https://v2.vuepress.vuejs.org/zh/reference/config.html#markdown.headers" and the correct link should be "https://v2.vuepress.vuejs.org/zh/reference/config. html#markdown-headers".

In order to facilitate other people who have the same doubts to find the answer, I will attach the solution to this problem:


1. config markdown in config.ts
    `export default defineUserConfig({
        markdown:{
              headers:{
                level: [2, 3, 4, 5]
              }
       },
}}`

2. config the frontmatter in the .md file
`---

headerDepth: 5

---`

**Reply by Mister-Hope** [reply]
1. PR welcomed for the link fix.
2. Your configuration is actually incorrect, and I believe a warning will appears in terminal. headerDepth 5 means h2,h3,h4,h5,h6, so you should include `6` in level, if you only want to deep down to h5, the depth is `4`

**Reply by hellobambi** [reply]
got it， thank you🙂
